tools/secret-rotation/Azure.Sdk.Tools.SecretRotation.Core/RotationPlan.cs

```diff
@@ -134,51 +134,67 @@ private async Task RotateAsync(string operationId, SecretState currentState, boo
         DateTimeOffset invocationTime = this.timeProvider.GetCurrentDateTimeOffset();
 
         SecretValue newValue = await OriginateNewValueAsync(operationId, invocationTime, currentState, whatIf);
+        // TODO: some providers will issue secrets for longer than we requested. Should we propagate the real expiration date, or the desired expiration date?
```

Comment:
Do we have a difference between expiration date and desired rotation date?

Reply:
we do have instances where an origin will issue a secret with an expiration date other than what was requested